File: Sorting Algorithms/InsertionSort/InsertionSort.js
//Insertion Sort in Python

function InsertionSort(array){
    for(let step=1;step<array.length;step++){
        let key=array[step]
        let j=step-1
        
        // Compare key with each element on the left of it until an element smaller than it is found
        // For descending order, change key<array[j] to key>array[j].  
        while((j>=0) && (array[j]> key)){
            array[j+1]=array[j]
            j--
        }

        //Place key at after the element just smaller than it.
        array[j+1]=key
    }
    return array
}


let array=[5,4,3,2,1]

console.log(array)
InsertionSort(array)
console.log(array)

File: Sorting Algorithms/InsertionSort/InsertionSort.py
#Insertion Sort in Python

def InsertionSort(arr):
    for step in range(1,len(arr)):
        key=arr[step]
        j=step-1

        # Compare key with each element on the left of it until an element smaller than it is found
        # For descending order, change key<array[j] to key>array[j].  
        while j>=0 and arr[j]>key:
            arr[j+1]=arr[j]
            j=j-1
            
        # Place key at after the element just smaller than it.    
        arr[j+1]=key 


arr=[5,4,3,2,1]
print(arr)
InsertionSort(arr)
print(arr)